scenario/LiquidationBotScenario.ts

```diff
@@ -18,46 +22,188 @@ enum Exchange {
   SushiSwap
 }
 
+async function borrowCapacityForAsset(comet: CometInterface, actor: CometActor, assetIndex: number) {
+  const {
+    asset: collateralAssetAddress,
+    borrowCollateralFactor,
+    priceFeed,
+    scale
+  } = await comet.getAssetInfo(assetIndex);
+
+  const userCollateral = await comet.collateralBalanceOf(
+    actor.address,
+    collateralAssetAddress
+  );
+  const price = await comet.getPrice(priceFeed);
+
+  const factorScale = await comet.factorScale();
+  const priceScale = await comet.priceScale();
+  const baseScale = await comet.baseScale();
+
+  const collateralValue = (userCollateral.mul(price)).div(scale);
+  return collateralValue.mul(borrowCollateralFactor).mul(baseScale).div(factorScale).div(priceScale);
+}
+
 for (let i = 0; i < MAX_ASSETS; i++) {
-  const borrowPositions = [
+  const assetAmounts = [
     // COMP
-    {
-      baseBorrowAmount: 20_000n,
-      assetAmount: ' == 1000'
-    },
+    ' == 500',
     // WBTC
-    {
-      baseBorrowAmount: 1_300_000n,
-      assetAmount: ' == 120'
-    },
+    ' == 120',
     // WETH
-    {
-      baseBorrowAmount: 5_000_000n,
-      assetAmount: ' == 5000'
-    },
+    ' == 5000',
     // UNI
-    {
-      baseBorrowAmount: 700_000n,
-      assetAmount: ' == 150000'
-    },
+    ' == 150000',
     // LINK
+    ' == 250000',
+  ];
+  scenario(
+    `LiquidationBot > liquidates an underwater position of $asset${i} ${assetAmounts[i] || ''} with no maxCollateralPurchase`,
     {
-      baseBorrowAmount: 1_000_000n,
-      assetAmount: ' == 250000'
+      filter: async (ctx) => ctx.world.base.network === 'mainnet' && await isValidAssetIndex(ctx, i),
+      tokenBalances: {
+        $comet: { $base: 1000 },
+      },
+      cometBalances: {
+        albert: {
+          [`$asset${i}`]: assetAmounts[i] || 0
+        },
+        betty: { $base: 1000 },
+      },
     },
+    async ({ comet, actors, assets }, _context, world) => {
+      const { albert, betty } = actors;
+      const { USDC } = assets;
+
+      const liquidator = await world.deploymentManager.deploy(
+        'liquidator',
+        'liquidator/Liquidator.sol',
+        [
+          RECIPIENT, // _recipient
+          UNISWAP_ROUTER, // _uniswapRouter
+          SUSHISWAP_ROUTER, // _sushiSwapRouter
+          comet.address, // _comet
+          UNISWAP_V3_FACTORY_ADDRESS, // _factory
+          WETH9, // _WETH9
+          10e6, // _liquidationThreshold,
+          [
+            '0xc00e94Cb662C3520282E6f5717214004A7f26888', // COMP
+            '0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599', // WBTC
+            '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2', // WETH
+            '0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984', // UNI
+            '0x514910771AF9Ca656af840dff83E8264EcF986CA', // LINK
```

Comment:
You can pull these from `assets` on line 76.

Reply:
updated